### PR TITLE
Tests identify, screen, group, flush, reset

### PR DESCRIPTION
**Pull request by ladanazita**
Tests basic functionality. `track` and tracking revenue was a bit more complex, so I will break that out into a separate PR for review

### CODE REVIEW

**Review by brennan — COMMENTED** [review]
I don't have any questions on these changes - but awaiting a pass from Prateek since these updates are pretty meaty.

**Comment by ladanazita** [comment]
@brennan That's fair - you can just view the test file to see what I'm testing for - there is no functionality change here

File: Example/Tests/Tests.m [hunk 1]
```diff
@@ -10,19 +10,112 @@
...
+            @"trackSessionEvents" : @true
+        } forAnalytics:nil];
+        expect(integration.settings).to.equal(@{ @"apiKey" : @"1234",
+                                                 @"trackSessionEvents" : @true });
+    });
+});
+
+describe(@"SEGAmplitudeIntegration", ^{
+
+    __block Amplitude *mockAmplitude;
```
Comment (f2prateek) [inline Example/Tests/Tests.m]: I think it's cleaner to just call this variable `amplitude`.

File: Pod/Classes/SEGAmplitudeIntegration.m [hunk 2]
```diff
@@ -11,12 +11,21 @@ - (id)initWithSettings:(NSDictionary *)settings
         self.settings = settings;
         self.amplitude = [Amplitude instance];
 
+        NSString *apiKey = [self.settings objectForKey:@"apiKey"];
+        [[Amplitude instance] initializeApiKey:apiKey];
```
Comment (f2prateek) [inline Pod/Classes/SEGAmplitudeIntegration.m]: This should be calling `initializeApiKey` on `self.amplitude`.

File: Pod/Classes/SEGAmplitudeIntegration.m [hunk 3]
```diff
@@ -11,12 +11,21 @@ - (id)initWithSettings:(NSDictionary *)settings
         self.settings = settings;
```
Comment (f2prateek) [inline Pod/Classes/SEGAmplitudeIntegration.m:11]: This method is not being tested. 

All this logic should be in the second initializer (the initWithSettings and andAmplitude one).

Then this method should call that second initializer, using `Amplitude instance` as the parameter for `andAmplitude`.

**Review by f2prateek — APPROVED** [review]
LGTM. Some room for improvement but can be tackled separately since this is already better than what we currently have.

File: Pod/Classes/SEGAmplitudeIntegration.m (hunk 3 above)
Comment (ladanazita) [reply Pod/Classes/SEGAmplitudeIntegration.m:11]: I made this change but the comment was on a line that didn't need to change

**Pull request merged**